src/controllers/mainController.ts

```diff
@@ -205,8 +205,14 @@ export default class MainController implements vscode.Disposable {
                 this._context.subscriptions.push(
                     vscode.commands.registerCommand(
                         Constants.cmdObjectExplorerNodeSignIn, async (node: AccountSignInTreeNode) => {
-                    this._objectExplorerProvider.signInNodeServer(node.parentNode);
-                    return this._objectExplorerProvider.refresh(undefined);
+                    let profile = <IConnectionProfile>node.parentNode.connectionCredentials;
+                    profile = await self.connectionManager.connectionUI.promptForRetryCreateProfile(profile);
```

Comment:
It looks like this is going to prompt every time? Wouldn't it be better to try and connect first and only if that fails then prompt for re-entering information?

Reply:
This is the command for the `Sign In` node. So it will only prompt if you click on it.

Reply:
I'm saying that it'll prompt you to go through the flow every time you click on the button. But if we have enough information stored to connect (either windows auth or correct username/password) we shouldn't go through that flow and instead just directly connect.

Reply:
Well, we don't know what's wrong so we do exactly what ADS does. Retry the profile with all the things pre-populated. We can't break flows in the extension because it doesn't have a dialog like ADS.

Reply:
This is the flow I'm imagining :

1. User clicks Sign In
2. We attempt to connect using the current credentials. If the connection is successful then great - we're done here (this could easily be the case if it was a transient error or something where the connection string was correct but something else happened)
3. Otherwise catch the error (and display it) then do the re-prompting flow like you have now

Does that make sense?

Reply:
The flow is exactly the same, except in the second part, we don't connect with the same credentials because we knew it was wrong the first time and hence got to the `Sign In` node. Otherwise, everything's the way you described.

Reply:
But as you said we don't know why we failed to connect. (Well...we do. We could be passing the actual error code back from the connection error. But that's beyond what we need to do here I think). So we shouldn't assume that it's something wrong with the connection information. If it's a transient issue (network down or whatever) then trying to do a reconnect may just work and save the user having to click enter a whole bunch of times.

Reply:
@Charles-Gagnon Error specific UX can be an enhancement for the next iteration, but for now I think this works fine. Would like to get this in for this release.

Reply:
Alright, although I don't think this should really be that much extra work and the experience as it is isn't great. But if you have other bugs to fix then just file an issue and get back to it later.

Reply:
I think the UX now is better than connecting to redundant incorrect credentials. Merging now and will look into better UX with PMs for later releases.